### PR TITLE
feat(plugin): allowing api-breaker to dynamically adapt to the BE rules

### LINKED ISSUE #1588: Optimize front-end validation of fields
## The current front-end approach
To create a route for example, the front-end will be regular [1] check for certain fields, such as name field, if the regular condition is not met, the front-end page will report an error and block the user's next operation.

image

## The disadvantages of the current approach
The current front-end can meet the verification requirements. However, when the back-end changes the verification rules, the front-end has to passively and manually update the verification rules for the corresponding fields. If the back-end does not notify the front-end in time or misses some fields, then the synchronization of the front-end and back-end verification will be a problem, affecting the user's use.

## Improvement measures
After communicating with @membphis , using jsonschema validation would be a good choice.
The front-end calls the API to get the jsonschema of the corresponding module and then maps the corresponding jsonschema rules to the corresponding field checks on the front-end. Do this when the back-end update the validation rules, the front-end can automatically synchronize the latest validation rules, reducing unnecessary development time of the front-end.

[1] regular expressions provided by the back-end, and then manually added by the front-end developers to the corresponding form form for the field validation

cc @juzhiyuan @liuxiran @guoqqqi

### PR DESCRIPTION
**Pull request by wmdmomo**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [ ] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

This PR will change the plug-in(api-breaker) under plugin/UI, so that the plug-in can dynamically adapt to the back-end json rules.

**Related issues**

fix/resolve #1588 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by LiteSun** [comment]
Hi, @wmdmomo, feel so happy to see your first PR !!!

I see that there are two plugins here in one PR, it would be better to split them into two. 😊

**Review by iamayushdas — APPROVED** [review]
LGTM, just split them

File: web/src/components/Plugin/UI/api-breaker.tsx [hunk 1]
```diff
@@ -39,14 +40,18 @@ const FORM_ITEM_WITHOUT_LABEL = {
...
 };
 
-const ApiBreaker: React.FC<Props> = ({ form }) => {
+const ApiBreaker: React.FC<Props> = ({ form, schema }) => {
   const { formatMessage } = useIntl()
-
+  const propertires = schema?.properties
+  const un_http_statuses = propertires.unhealthy.properties.http_statuses
+  const un_http_default = un_http_statuses.default
+  const {http_statuses} = propertires.healthy.properties
```
Comment (LiteSun) [inline web/src/components/Plugin/UI/api-breaker.tsx:48]:
```suggestion
  const { http_statuses } = propertires.healthy.properties
```

**Comment by iamayushdas** [comment]
@wmdmomo can i have your slack id?

**Review by iamayushdas — APPROVED** [review]

**Comment by guoqqqi** [comment]
Hi, Thank you for your contribution. Here are some conflicts in this PR that need to be resolved

**Comment by juzhiyuan** [comment]
@wmdmomo Please resolve your conflicts, thansk

**Comment by wmdmomo** [comment]
@juzhiyuan I have resolved the conflicts.

**Review by guoqqqi — APPROVED** [review]

**Review by Baoyuantop — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Review by LiteSun — APPROVED** [review]

**Review by liuxiran — APPROVED** [review]

**Pull request merged**